bundles/com.espressif.idf.launch.serial.core/src/com/espressif/idf/launch/serial/util/EspFlashCommandGenerator.java

```diff
@@ -0,0 +1,40 @@
+/*******************************************************************************
+ * Copyright 2021 Espressif Systems (Shanghai) PTE LTD. All rights reserved.
+ * Use is subject to license terms.
+ *******************************************************************************/
+
+package com.espressif.idf.launch.serial.util;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.eclipse.core.runtime.preferences.InstanceScope;
+import org.osgi.service.prefs.Preferences;
+
+import com.espressif.idf.core.IDFConstants;
+import com.espressif.idf.core.util.IDFUtil;
+
+public class EspFlashCommandGenerator {
+	/**
+	 * @param launch
+	 * @return command to flash the application
+	 */
+	public static String getEspFlashCommand() {
+
+		List<String> commands = new ArrayList<>();
+		commands.add(IDFUtil.getIDFPythonScriptFile().getAbsolutePath());
+		commands.add("-p"); //$NON-NLS-1$
+
+		String serialPort = getLastUsedSerialPort();
+		commands.add(serialPort);
+
+		commands.add(IDFConstants.FLASH_CMD);
+
+		return String.join(" ", commands); //$NON-NLS-1$
+	}
+
+	private static String getLastUsedSerialPort() {
+		Preferences preferences = InstanceScope.INSTANCE.getNode("com.espressif.idf.launch.serial.ui"); //$NON-NLS-1$
+		return preferences.get("com.espressif.idf.launch.serial.core.serialPort", ""); //$NON-NLS-1$ //$NON-NLS-2$
+	}
```

Comment:
Use SerialFlashLaunchTargetProvider.ATTR_SERIAL_PORT

Reply:
@kolipakakondal new commit with requested changes:
https://github.com/espressif/idf-eclipse-plugin/pull/294/commits/e5a519a9b2433b2558df586078d9897a46545f92

Now EspFlashCommandGenerator takes serialPort string, which is taken from SerialFlashLaunchTargetProvider.ATTR_SERIAL_PORT attribute